### PR TITLE
Add Ender pearls

### PR DESCRIPTION
**Pull request by JackNoordhuis**
### Description
Add the basic vanilla Ender Pearl implementation

### Reason to modify
Ender Pearls are a vanilla feature commonly used by servers.

### Tests & Reviews
Thoroughly tested by myself, works exactly like vanilla.

### CODE REVIEW

**Comment by Pizzahaet** [comment]
I love you for this (no homo)

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -2524,6 +2528,45 @@ public function handleDataPacket(DataPacket $packet){
 						}else{
 							$thrownPotion->spawnToAll();
 						}
+					}elseif($item->getId() === Item::ENDER_PEARL){
+						if(floor(($time = microtime(true)) - $this->lastEnderPearlUse) >= 1){
```
Comment (ishitatsuyuki) [inline src/pocketmine/Player.php]: Floor seems redundant to me.

**Comment by diamond-gold** [comment]
Bug: if ProjectileLaunchEvent is cancelled for ender pearl, it damages the player and teleports the player to the same position

**Comment by JackNoordhuis** [comment]
The projectile bug seems to happen with thrown potions as well, this should be fixed in a separate commit as this pull request is simply to add ender pearls, not fix an existing projectile bug. @diamond-gold 

@ishitatsuyuki I can remove the floor if you like, I add it to all my micro time comparisons to make them more accurate.

**Comment by ishitatsuyuki** [comment]
How can it mathematically make it more accurate ..?

**Comment by JackNoordhuis** [comment]
In some sense it's actually less accurate, but we only want the check to pass if it's been more than x seconds so anything less doesn't matter. Rounding the value to the lowest whole number was faster than comparing float's in PHP5 but I just did some benchmark tests that have shown this to be the opposite with PHP7.

**Comment by diamond-gold** [comment]
@JackNoordhuis The bug is because when you cancel ProjectileLaunchEvent, you called $enderpearl->kill()
And inside Enderpearl::kill() you damage the player and teleport the player, it is not an existing bug

**Comment by JackNoordhuis** [comment]
@diamond-gold I'm aware of what's causing the issue, I've been busy cleaning up the projectile code as a whole. A fix will be applied soon.

**Pull request merged**